pylint/checkers/base.py

```diff
@@ -1955,6 +1972,14 @@ def _name_invalid_due_to_blacklist(self, name: str) -> bool:
     def _check_name(self, node_type, name, node, confidence=interfaces.HIGH):
         """check for a name using the type's regexp"""
 
+        non_ascii_regexp = re.compile("[^\u0000-\u007F]")
```

Comment:
Can you move the compilation at `__init__` level? This way we don't have to compile a regex for every name we analyze.

Reply:
done!